### PR TITLE
feat: homework tasks init

### PR DESCRIPTION
**Pull request by KubaKlin**

### CODE REVIEW

File: scripts.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    if (currentNumber < smallestNumber){
+      smallestNumber = currentNumber;
+    }
+  }
+
+  return smallestNumber;
+}
+
+function getSmallestNumberSort(numbers) {
+  numbers.sort(function(a, b){return a - b});
```
Comment (mwanago) [inline scripts.js]: Don't use one-letter variables. Please avoid repeating the same mistake in multiple PRs

Comment (mwanago) [inline scripts.js]: Don't declare a function in just one line. Use multiple lines

File: scripts.js [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+  return characters.join('');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+//5
+function isPalindrome(string) {
+  string = string.replace(/\s/g, '');
```
Comment (mwanago) [inline scripts.js:76]: You can use `replaceAll` instead

File: scripts.js [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+  characters = characters.split('');
+
+  for (let character of characters) {
```
Comment (mwanago) [inline scripts.js]: Use a regular `for` loop

File: scripts.js [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+  return countedLetters;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(peopleArray) {
```
Comment (mwanago) [inline scripts.js:111]: The `getYoungestPerson` sounds like it is supposed to return the entire information about a person. In your case it doesn't

File: scripts.js [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  let youngestPerson = 100;
```
Comment (mwanago) [inline scripts.js]: Why 100?

File: scripts.js [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+  for (let i = 0; i < peopleArray.length; i++) {
+    if (youngestPerson > peopleArray[i].age) {
+      youngestPerson = peopleArray[i].age;
+    }
+  }
+
+  return youngestPerson;
+}
+
+function getOldestPerson(peopleArray) {
```
Comment (mwanago) [inline scripts.js:123]: The `getOldestPerson` sounds like it is supposed to return the entire information about a person. In your case, it doesn't

File: scripts.js [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+// 8
+function positiveSum(numbers) {
+  let numbersSummary = 0;
```
Comment (mwanago) [inline scripts.js]: It's a sum, not really a "summary".

File: scripts.js [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+  }
+
+  return numbersSummary;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, -5, -10]));
+
+// 9
+function squaredNumbersSummary(numbers) {
```
Comment (mwanago) [inline scripts.js:179]: It's a sum, not really a "summary".

File: scripts.js [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+    total = total + numbers[i] * numbers[i];
+  }
+
+  return total;
+}
+
+console.log(squaredNumbersSummary([0, 3, 4, 5]));
+
+// 10
+function abbrevName(name){
```
Comment (mwanago) [inline scripts.js]: Don't use abbreviations. Try not making this mistake going further

File: scripts.js [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+  let initials = name[0] + '.';
+
+  for (let i = 0; i < name.length; i++) {
+    if (name[i] === ' ') {
```
Comment (mwanago) [inline scripts.js]: It would be simpler to use `split`

File: scripts.js [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+  }
+
+  return initials.toUpperCase();
+}
+
+console.log(abbrevName('Kuba klin'));
+
+// 11
+function countSheeps(sheep) {
+  let presenetSheep = 0;
```
Comment (mwanago) [inline scripts.js]: I think there is a typo here

File: scripts.js [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+
+  for (let i = 0; i < sheep.length; i++) {
```
Comment (mwanago) [inline scripts.js]: Why `i++` and not `++i`?

File: scripts.js [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+    }
+  }
+
+  return presenetSheep;
+}
+
+// 12
+function digitize(numbers) {
+  let reverseArray = [];
+  let digits = ('' + numbers).split('').map(Number);
```
Comment (mwanago) [inline scripts.js]: Don't use `map` yet

File: scripts.js [hunk 14]
```diff
@@ -0,0 +1,323 @@
...
+}
+
+console.log(doubleNumbers([1, 2, 9]));
+
+// 15
+function invertNumbers(numbers) {
+  let invertedNumbers = [...numbers];
+
+  for (let i = 0; i < invertedNumbers.length; i++) {
+    invertedNumbers[i] = -invertedNumbers[i];
```
Comment (mwanago) [inline scripts.js]: Use `* -1` for clarity instead

File: scripts.js [hunk 15]
```diff
@@ -0,0 +1,323 @@
...
+function summary(numbers) {
+  let summaryNumbers = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (typeof numbers[i] === 'number') {
+      summaryNumbers = summaryNumbers + numbers[i]
+    }
+  }
+
+  return summaryNumbers
```
Comment (mwanago) [inline scripts.js]: Don't forget about semicolons. Try not making the same mistake in multiple PRs

File: scripts.js [hunk 16]
```diff
@@ -0,0 +1,323 @@
...
+  return invertedNumbers;
+}
+
+// 16
+function summary(numbers) {
+  let summaryNumbers = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (typeof numbers[i] === 'number') {
+      summaryNumbers = summaryNumbers + numbers[i]
```
Comment (mwanago) [inline scripts.js]: Don't forget about semicolons. Try not making the same mistake in multiple PRs

File: scripts.js [hunk 17]
```diff
@@ -0,0 +1,323 @@
...
+
+  return invertedNumbers;
+}
+
+// 16
+function summary(numbers) {
+  let summaryNumbers = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (typeof numbers[i] === 'number') {
```
Comment (mwanago) [inline scripts.js]: The exercise states that "You can assume that you are only given numbers."

File: scripts.js [hunk 18]
```diff
@@ -0,0 +1,323 @@
...
+
+// 19
+function getAverage(marks){
+  let averageMarks = 0;
+
+  for (let i = 0; i < marks.length; i++) {
+    averageMarks += marks[i];
+  }
+
+  return Math.floor((averageMarks / marks.length));
```
Comment (mwanago) [inline scripts.js]: There are too many round brackets here

File: scripts.js [hunk 19]
```diff
@@ -0,0 +1,327 @@
...
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+
+  for (let i = 0; i < characters.length; ++i){
+    let character = characters.charAt(i)
+    if (!countedLetters[character]) {
+      countedLetters[character] = 1;
+    }
+    else {
```
Comment (mwanago) [inline scripts.js]: `}` and `else {` should be in the same line
```
if (...) {
  ...
} else {
  ...
}
```

File: scripts.js [hunk 20]
```diff
@@ -0,0 +1,327 @@
...
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
```
Comment (mwanago) [inline scripts.js]: There is a space missing here

File: scripts.js [hunk 21]
```diff
@@ -0,0 +1,327 @@
...
+    total = total + numbers[i] * numbers[i];
+  }
+
+  return total;
+}
+
+console.log(squaredNumbersSummary([0, 3, 4, 5]));
+
+// 10
+function abbreviationName(name) {
```
Comment (mwanago) [inline scripts.js]: `abbreviation` is a noun. Since a function is supposed to _do_ something, use a verb instead.

File: scripts.js [hunk 22]
```diff
@@ -0,0 +1,327 @@
...
+  let words = name.split(/\s+/);
+  let abbreviation = '';
+  for (let i = 0; i < words.length; ++i) {
+    abbreviation += words[i][0] + '.';
+  }
+  return abbreviation.toUpperCase().substring(0, abbreviation.length - 1);
```
Comment (mwanago) [inline scripts.js]: You can do that in a much simpler way.
```
const nameParts = name.split(' ');
const firstName = nameParts[0].toUpperCase();
const lastName = nameParts[1].toUpperCase();

return ...
```

File: scripts.js [hunk 23]
```diff
@@ -0,0 +1,327 @@
...
+      presenetSheep = presenetSheep + 1;
+    }
+  }
+
+  return presenetSheep;
+}
+
+// 12
+function digitize(numbers) {
+  let reverseArray = [];
```
Comment (mwanago) [inline scripts.js]: No need to use `let` here.

File: scripts.js [hunk 24]
```diff
@@ -0,0 +1,327 @@
...
+  let digits = ('' + numbers).split('').reverse();
```
Comment (mwanago) [inline scripts.js]: No need to use `let` here.

Comment (mwanago) [inline scripts.js]: Why `'' + numbers`? Please explain.

Comment (KubaKlin) [reply scripts.js]: if I understand it correctly, to turn the number into the string

File: scripts.js (hunk 12 above)
Comment (KubaKlin) [reply scripts.js]: typo. Didn't know there is a difference between ++i and i++

File: scripts.js (hunk 5 above)
Comment (KubaKlin) [reply scripts.js]: to compare the numbers downwards

File: scripts.js [hunk 25]
```diff
@@ -0,0 +1,324 @@
...
+    }
+  }
+
+  return presenetSheep;
+}
+
+// 12
+function digitize(numbers) {
+  reverseArray = [];
+  digits = ('' + numbers).split('').reverse();
```
Comment (mwanago) [inline scripts.js]: Use `const`

File: scripts.js (hunk 24 above)
Comment (mwanago) [reply scripts.js]: That's correct. However, this is a very **implicit** way to convert a number into a string. It's difficult to read and might be confusing.
Instead, do `String(numbers)`. It's always better to be explicit if possible.

File: scripts.js (hunk 4 above)
Comment (KubaKlin) [reply scripts.js:111]: You mean the name of the function? That is the given name in homework task. Should I change it?

File: scripts.js (hunk 6 above)
Comment (KubaKlin) [reply scripts.js:123]: You mean the name of the function? That is the given name in homework task. Should I change it?

File: scripts.js (hunk 11 above)
Comment (KubaKlin) [reply scripts.js]: I thought that it was about the plural name of word 'sheep'. Fixed both now

File: scripts.js (hunk 24 above)
Comment (KubaKlin) [reply scripts.js]: ok, fixed

File: scripts.js (hunk 4 above)
Comment (mwanago) [reply scripts.js:111]: No. The name `getYoungestPerson` sounds like it is supposed to return all the information about a person. It means you should return the object containing the youngest person, not just their age.

File: scripts.js (hunk 6 above)
Comment (mwanago) [reply scripts.js:123]: No. The name `getOldestPerson` sounds like it is supposed to return all the information about a person. It means you should return the object containing the oldest person, not just their age.

Comment (KubaKlin) [reply scripts.js:123]: ok, now I get it - fixed

File: scripts.js (hunk 4 above)
Comment (KubaKlin) [reply scripts.js:111]: ok, now I get it - fixed

**Review by mwanago — APPROVED** [review]

**Pull request merged**